borg/testsuite/convert.py

```diff
@@ -0,0 +1,97 @@
+import os
+import pytest
+import shutil
+import tempfile
+
+try:
+    import attic.repository
+    import attic.key
+    import attic.helpers
+except ImportError:
+    attic = None
+pytestmark = pytest.mark.skipif(attic is None,
+                                reason = 'cannot find an attic install')
+
+from ..converter import AtticRepositoryConverter, NotImplementedException, AtticKeyfileKey
+from ..helpers import get_keys_dir
+from ..key import KeyfileKey
+from ..repository import Repository, MAGIC
+from . import BaseTestCase
+
+class ConversionTestCase(BaseTestCase):
+
+    def open(self, path, repo_type  = Repository, create=False):
+        return repo_type(os.path.join(path, 'repository'), create = create)
+
+    def setUp(self):
+        self.tmppath = tempfile.mkdtemp()
+        self.attic_repo = self.open(self.tmppath,
+                                    repo_type = attic.repository.Repository,
+                                    create = True)
+        # throw some stuff in that repo, copied from `RepositoryTestCase.test1`_
+        for x in range(100):
+            self.attic_repo.put(('%-32d' % x).encode('ascii'), b'SOMEDATA')
+        self.attic_repo.close()
+
+    def tearDown(self):
+        shutil.rmtree(self.tmppath)
+
+    def check_repo(self, state = True):
```

Comment:
maybe this could rather be repo_valid(self) and return the state of the repo.

then the assert is done in the test method itself (as expected) and would read nicely like:
```
    assert repo_valid()
    # or
    assert not repo_valid()
```